src/main/java/net/greatstart/controllers/UserController.java

```diff
@@ -62,10 +78,35 @@ public Principal user(Principal principal) {
             @Valid @RequestBody DtoUserProfile dtoUser) {
         DtoUserProfile currentDtoUser = userService.updateUser(dtoUser, id);
         if (currentDtoUser != null) {
+            currentDtoUser.setInitial(getInitials(currentDtoUser.getName(), currentDtoUser.getLastName()));
             return new ResponseEntity<>(currentDtoUser, HttpStatus.OK);
         }
         return new ResponseEntity<>(HttpStatus.NOT_FOUND);
 
     }
 
+    @PostMapping("/api/user")
+    public ResponseEntity<DtoUser> processRegistration(@Valid @RequestBody DtoUser user) {
+
+        if (userService.getUserByEmail(user.getEmail()) == null) {
+            userService.createUser(user.getEmail(), passwordEncoder.encode(user.getPassword()));
+            securityService.autoLogin(user.getEmail(), user.getPassword());
+            return new ResponseEntity<DtoUser>(HttpStatus.OK);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)